sway-core/src/type_system/type_parameter.rs

```diff
@@ -137,22 +139,50 @@ impl TypeParameter {
             );
         }
 
-        // Insert the type parameter into the namespace as a dummy type
-        // declaration.
-        let type_parameter_decl = ty::TyDeclaration::GenericTypeForFunctionScope {
-            name: name_ident.clone(),
-            type_id,
-        };
-        ctx.namespace
-            .insert_symbol(name_ident.clone(), type_parameter_decl)
-            .ok(&mut warnings, &mut errors);
+        if is_from_parent {
```

Comment:
This function is only called when type parameters are declarated. i.e. when we see `fn my_function<T, E>(..)` and `impl<V> Data<V> { .. }`, is that what you are intending? Otherwise, the "else" path that you have here is how the type parameters are distributed to the other instances in which those type parameters are used.